common/cache/namespaceCache.go

```diff
@@ -776,26 +776,27 @@ var SampleRetentionKey = "sample_retention_days"
 // SampleRateKey is key to specify sample rate
 var SampleRateKey = "sample_retention_rate"
 
-// GetRetentionDays returns retention in days for given workflow
-func (entry *NamespaceCacheEntry) GetRetentionDays(
+// GetRetention returns retention in days for given workflow
+func (entry *NamespaceCacheEntry) GetRetention(
 	workflowID string,
-) int32 {
+) time.Duration {
+
+	if entry.config.Retention == nil {
+		return 0
+	}
 
 	if entry.IsSampledForLongerRetention(workflowID) {
 		if sampledRetentionValue, ok := entry.info.Data[SampleRetentionKey]; ok {
```

Comment:
Yes, I didn't change that part, because I couldn't figure out (in a short amount of time) where it came from. It would be easy to use the same `timestamp.ParseDurationDefaultDays` function here and then it could support other units and also be backwards compatible. Should I change that? And then, should I change the key also? (It's `sample_retention_days`)

Reply:
The sample retention days was introduced (long ago) specifically as a short term workaround before long term archival was available. Do we still need sample retention nowadays? @wxing1292 Maybe we should get ride of it entirely.

Reply:
@dnr , to answer your question on where this data is coming from, it is supplied when you create / update namespace as custom supplied data: 
--namespace_data value, --dmd value             Namespace data of key value pairs, in format of k1:v1,k2:v2,k3:v3

Reply:
This PR is good to go. We should do a follow up to clean up sampled retention: #1622